### PR TITLE
Added Aspect entity and usage of aspects by Opportunity

### PR DESCRIPTION
**Pull request by techsmyth**
New entity: Aspect
Opportunity updated to have an array of aspects
new mutation for adding an aspect to an Opportunity

### CODE REVIEW

File: src/domain/opportunity/opportunity.entity.ts [hunk 1]
```diff
@@ -69,6 +70,17 @@ export class Opportunity extends BaseEntity
...
   projects?: Project[];
 
+  @Field(() => [Aspect], {
+    nullable: true,
+    description: 'The set of solution aspects for this Opportunity',
+  })
+  @OneToMany(
+    () => Aspect,
+    aspect => aspect.opportunity,
+    { eager: true, cascade: true }
```
Comment (valentinyanakiev) [inline src/domain/opportunity/opportunity.entity.ts:80]: Same comment as other PRs - do we want lazy loading?

**Review by valentinyanakiev — COMMENTED** [review]
Overall OK, the only comment is about eager loading.

File: src/domain/opportunity/opportunity.entity.ts (hunk 1 above)
Comment (techsmyth) [reply src/domain/opportunity/opportunity.entity.ts:80]: same comment as on actor groups / actor - lets first get it working and then can see re not using eager loading.

**Review by valentinyanakiev — APPROVED** [review]

**Comment by techsmyth** [comment]
Lets go with the eager loading for now, if it does have an impact then will optimise. For now key to get the domain model extended so we can discuss its usage with Momentum.

**Pull request merged**